c/include/proton/annotations.h

```diff
@@ -90,4 +90,24 @@
     #endif
 #endif
 
+// fallthrough
+#if defined __cplusplus && defined __has_cpp_attribute
+    #if __has_cpp_attribute(fallthrough) && __cplusplus >= __has_cpp_attribute(fallthrough)
+        #define PN_FALLTHROUGH [[fallthrough]]
+    #endif
+#endif
+#if !defined PN_FALLTHROUGH && defined __STDC_VERSION__ && defined __has_c_attribute
+    #if __has_c_attribute(fallthrough) && __STDC_VERSION__ >= __has_c_attribute(fallthrough)
+        #define PN_FALLTHROUGH [[fallthrough]]
+    #endif
+#endif
+#if !defined PN_FALLTHROUGH && defined __has_attribute
+    #if __has_attribute(__fallthrough__)
+        #define PN_FALLTHROUGH __attribute__((__fallthrough__))
+    #endif
+#endif
+#if !defined PN_FALLTHROUGH
+    #define PN_FALLTHROUGH (void)0
```

Comment:
Although does this actually tell a linter that there is an expected fallthrough? - That is what the `//Fallthru` comments are for, but I don't think you can preprocess into comments.

Reply:
True, this depends on all relevant linters knowing one of the other ways to annotate the fallthrough.

The default is there for the case of a compiler that does not know how to warn about missed fallthrough but knows to warn about extraneous semicolon. Not that I'd expect to ever encounter something like this.